djweb/simulations/static/js/ibm704sim.js

```diff
@@ -930,25 +942,64 @@ class IBM_704 {
     assemble(origin, code_lines) {
         let register = origin;
         for (let line_no in code_lines) {
+            if (register >= this.size || register < 0) {
+                alert("Error: Tried to program to invalid register on line "
+                + (parseInt(line_no)+1) + "!  Register must be between 0 and " + (this.size-1) + ".");
+                throw INVALID_REGISTER_EXCEPTION;
+            }
             let line = code_lines[line_no];
             console.log(line);
             let operation = line.substring(0,3);
```

Comment:
The problem with this split is that it thinks `STOO 4` is valid. 
You could use regular expressions to parse the input, e.g. (you can mess around with regex here: https://rubular.com/)
```
let regex_line_parser = new RegExp('^([A-Z]{3}) ([0-9, ]+)')
let parsed_command = regex_line_parser.exec(line)
//if parsed command is null, throw error, not a valid command.
let operation = parsed_command[1]
let rest_of_line = parsed_command[2]
```